deluge/plugins/AutoAdd/deluge_autoadd/core.py

```diff
@@ -340,6 +340,8 @@ def fail_torrent_add(err_msg, filepath, magnet):
                     d = component.get('Core').add_torrent_magnet(
                         filedump.strip(), options
                     )
+                    d.addCallback(on_torrent_added, filename, filepath)
+                    d.addErrback(fail_torrent_add, filepath, magnet)
```

Comment:
Instead of duplicating here, unindent line 349 and 350 to add the callbacks to either deferred